Nadezhda Gubenko

nadezhdagub opened the pull request:
class implementation PopularMap

Commit:
Nadezhda Gubenko
class implementation PopularMap

11cfif commented:
название домашки кто будет писть?

11cfif commented on src/main/java/ru/mail/polis/homework/collections/PopularMap.java:
```diff
@@ -14,130 +10,187 @@
     private final Map<K, V> map;
+    private final Map<K, Integer> popularKey;
+    private final Map<V, Integer> popularValue;
+    K popularK;
```
а что с видимостью

nadezhdagub replied:
поставила private

11cfif commented on src/main/java/ru/mail/polis/homework/collections/PopularMap.java:
```diff
@@ -14,130 +10,187 @@
-        return null;
+        addPopularityKey((K) key);
+        V value = map.remove(key);
+        addPopularityValue(value);
```
а тут проверка на нал?

nadezhdagub replied:
исправила

11cfif commented on src/main/java/ru/mail/polis/homework/collections/PopularMap.java:
```diff
@@ -14,130 +10,187 @@
-        return null;
+        List<V> list = new ArrayList<>();
+
+        for (Map.Entry<V, Integer> entry : popularValue.entrySet()) {
```
хэшмапа дает гарантии на порядок?

nadezhdagub replied:
нет, не дает. Добавила sort

11cfif commented on src/main/java/ru/mail/polis/homework/collections/PopularMap.java:
```diff
@@ -14,130 +10,187 @@
+        if (key == null) {
+            return popularK;
+        }
+        popularKey.put(key, popularKey.getOrDefault(key, 0) + 1);
```
у вас в этом методе 4 обращения к мапе - можно свести до 2

nadezhdagub replied:
объединила методы, сократила обращения

11cfif commented on src/main/java/ru/mail/polis/homework/collections/PopularMap.java:
```diff
@@ -14,130 +10,187 @@
+        return popularK;
+    }
+
+    private V addPopularityValue(V values) {
```
можно ли с помощью дженериков объединить два метода?

nadezhdagub replied:
можно, объединила

Commit:
Nadezhda Gubenko
Homework 4, second fix

Commit:
Nadezhda Gubenko
Homework 4, third fix. Fix Iterator method with streams. Other fixes are in the last commit

11cfif commented on src/main/java/ru/mail/polis/homework/collections/PopularMap.java:
```diff
@@ -14,130 +10,167 @@
+    }
+
+    private <T> T addPopularityKeyOrValue(Map<T, Integer> countersMap, T popularEl, T element) {
+        if (element == null) {
```
а зачем еще сверху проверяете на налл (перед вызовом этого метода)

11cfif commented on src/main/java/ru/mail/polis/homework/collections/PopularMap.java:
```diff
@@ -14,130 +10,167 @@
      */
     public Iterator<V> popularIterator() {
-        return null;
+        return popularValue.entrySet()
```
после дедлайна
